packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -1,60 +1,93 @@
-import {
-  BaseLlmParameters,
-  CustomRequestConfig,
-  executeRequest
-} from '@sap-ai-sdk/core';
-import { BaseClient } from '../interface.js';
+import { HttpRequestConfig } from '@sap-cloud-sdk/http-client';
+import { CustomRequestConfig, executeRequest } from '@sap-ai-sdk/core';
+import { DeploymentResolver, resolveDeployment } from '../../core.js';
 import {
   OpenAiChatCompletionParameters,
   OpenAiEmbeddingParameters,
   OpenAiEmbeddingOutput,
-  OpenAiChatCompletionOutput
+  OpenAiChatCompletionOutput,
+  OpenAiChatModel,
+  OpenAiEmbeddingModel
 } from './openai-types.js';
 
 const apiVersion = '2024-02-01';
 
 /**
  * OpenAI GPT Client.
  */
-export class OpenAiClient implements BaseClient<BaseLlmParameters> {
```

Comment:
see https://github.com/SAP/ai-sdk-js/pull/50#discussion_r1702010223

Reply:
I believe the purpose of the `BaseClient` was rather for future use, so that other clients of e.g. Gemini / Claude etc. all align with the same base client, do you think this would make sense to keep / help with the implementation of additional clients?

Reply:
(just for completeness) In addition to YAGNI, [this change](https://github.com/SAP/ai-sdk-js/pull/75/files#diff-3b1b4fce21b3de60726e49eea5f4559cf76699d393e75495f681ffdee58d62deL57) is related. The `executeRequest` is also used for non-inference calls. There deployment ID doesn't make any sense (which is why I am removing it), and consequently I am removing the `<Data extends BaseLlmParameters>` on that method, which was the only usage.